### ISSUE TITLE
Understanding the `norm` factor before `weighted_cross_entropy_with_logits`

### ISSUE THREAD

**Issue by ClaudMor** [deezer/gravity_graph_autoencoders#13]
Hello,

First of all, thanks for your well-written and interesting article, code availability and continued support here on GitHub.

I currently don't understand the role of the `norm` term before the `weighted_cross_entropy_with_logits` (like in [this line](https://github.com/deezer/gravity_graph_autoencoders/blob/fa098943ad8b25de14882d76f3ad08253b30ab35/gravity_gae/optimizer.py#L36)). In particular:

1. Why should it improve training in the Gravity-GAE model (where it only rescales the loss)?
2. Why does it evenly balance the contributions of the  `weighted_cross_entropy_with_logits` term and the Kullback-Leibler term in the Gravity-VGAE loss ?

In my testing in a (yet to be perfected) torch_geometric-based attempt to reproduce your results, the `norm` term slightly improves final performance also for the Gravity-GAE model, where it should only act as a linear scaling for the loss and thus (in my understanding) shouldn't matter.

I googled "weighted cross entropy normalization" but I found other types of normalization I cannot immediately identify with yours.

**Comment by GuillaumeSalhaGalvan** [comment]
Dear Claudio,

This `norm` term was already included in the [original GAE/VGAE implementations](https://github.com/tkipf/gae).

I'd need to dive back into this code in the next few days to give a definitive answer. 
But, from a first quick look:

1. I'd agree that `norm` acts as a simple rescaling factor for the non-variational GAE;
2. YH-UtMSB explains the intuition behind `norm` [in this related issue](https://github.com/tkipf/gae/issues/17).

I hope this helps!

Guillaume

**Comment by ClaudMor** [comment]
Hi Guillame,

Thanks, I missed that!

Of course there's no need for you to go through the details, I'll read it up myself!

Definitively helped.

**Comment by GuillaumeSalhaGalvan** [comment]
Good luck with your project!
